From f97aaf80dd44e92f2cabc7e6d92d461f4fe6eddd Mon Sep 17 00:00:00 2001
From: Antonio Ospite <ao2@ao2.it>
Date: Tue, 16 Oct 2018 18:42:20 +0200
Subject: [PATCH 36/53] eval: Silence compiler warning about missing
 parentheses
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Gcc gives a warning about some missing parentheses:

-----------------------------------------------------------------------
eval.c: In function ‘evaltree’:
eval.c:282:15: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
   if (!status == isor || evalskip)
               ^~
eval.c:282:7: note: add parentheses around left hand side expression to silence this warning
   if (!status == isor || evalskip)
       ^~~~~~~
       (      )
-----------------------------------------------------------------------

Add the parentheses to silence the warning.

Signed-off-by: Antonio Ospite <ao2@ao2.it>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
---
 src/eval.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/eval.c b/src/eval.c
index dde9fa2..943948a 100644
--- a/src/eval.c
+++ b/src/eval.c
@@ -279,7 +279,7 @@ checkexit:
 		isor = n->type - NAND;
 		status = evaltree(n->nbinary.ch1,
 				  (flags | ((isor >> 1) - 1)) & EV_TESTED);
-		if (!status == isor || evalskip)
+		if ((!status) == isor || evalskip)
 			break;
 		n = n->nbinary.ch2;
 evaln:
-- 
2.25.0

